Documentation/Design/Classes/Class Declarations.md

```diff
@@ -0,0 +1,121 @@
+# Class Declarations
+
+## Feature Overview
+
+This improvement adds support for class declarations to GLS, including:
+
+* Naming a class
+* Declaring any generics *(optional)*
+* Declaring a class to extend *(optional)*
+
+
+## Language Examples
+
+### Java  
+```java
+class Point
+{
+}
+
+class Measurements<T>
+{
+}
+
+class Shape extends Measurements<Point>
```

Comment:
Nit: Why, in Java's example, does `Shape` inherit from `Measurements<Point>`, but in C# and TypeScript, `Shape` inherits from `int` and `number`?